# Week 11: Introduction to Distributions and Their Use

This notebook answers the questions from week 11 homework

In [1]:
import numpy as np
from scipy import stats

# Population generation from the prompt
np.random.seed(28)
dBP = np.random.normal(85, 10, 100000)

# Population mean check
population_mean = dBP.mean()
population_mean

np.float64(84.95549142317856)

In [2]:
# Use random seed value of 12 for the tasks below
np.random.seed(12)

# 1) Random sample of 40
sample = np.random.choice(dBP, size=40, replace=False)

# 2) Sample mean
sample_mean = sample.mean()

# 3) Sample standard deviation (sample SD uses ddof=1)
sample_std = sample.std(ddof=1)

# 4) Standard error
standard_error = sample_std / np.sqrt(len(sample))

# 5) Confidence coefficient for 95% CI with df = n - 1
c = stats.t.ppf(0.975, df=len(sample) - 1)

# 6) Lower confidence bound
lower_bound = sample_mean - c * standard_error

# 7) Upper confidence bound
upper_bound = sample_mean + c * standard_error

# 8) 1000 bootstrapped sample means, sorted ascending
boot_means = np.empty(1000)
for i in range(1000):
    bs = np.random.choice(sample, size=40, replace=True)
    boot_means[i] = bs.mean()
bootstrapped_means = np.sort(boot_means)

# 9) 2.5th percentile
percentile_2_5 = np.percentile(bootstrapped_means, 2.5)

# 10) 97.5th percentile
percentile_97_5 = np.percentile(bootstrapped_means, 97.5)

results = {
    "sample_mean": sample_mean,
    "sample_std": sample_std,
    "standard_error": standard_error,
    "c": c,
    "lower_bound": lower_bound,
    "upper_bound": upper_bound,
    "percentile_2_5": percentile_2_5,
    "percentile_97_5": percentile_97_5
}
results

{'sample_mean': np.float64(85.18752155840876),
 'sample_std': np.float64(9.627818569777178),
 'standard_error': np.float64(1.5222917789680321),
 'c': np.float64(2.022690920036761),
 'lower_bound': np.float64(82.10839579944351),
 'upper_bound': np.float64(88.26664731737401),
 'percentile_2_5': np.float64(82.20259484930493),
 'percentile_97_5': np.float64(88.1769982523287)}

## Expected Numeric Results

Using the exact setup above:

- `sample_mean` = 85.18752155840876
- `sample_std` = 9.627818569777178
- `standard_error` = 1.5222917789680321
- `c` = 2.022690920036761
- `lower_bound` = 82.10839579944351
- `upper_bound` = 88.26664731737401
- 2.5th percentile of `bootstrapped_means` = 82.20259484930493
- 97.5th percentile of `bootstrapped_means` = 88.1769982523287